wfl/Controllers/RestaurantsController.cs

```diff
@@ -23,33 +23,43 @@ public RestaurantsController(IDBManager dbManager)
             {
                 return BadRequest();
             }
-            Restaurant r = await DBManager.InsertRestaurantAsync(restaurant);
+            Restaurant r = new Restaurant()
+            {
+                Name = restaurant.Name,
+                MenuLink = restaurant.MenuLink
+            };
+            r = await DBManager.InsertEntityAsync(r);
             return Created(r.Id.ToString(), r);
         }
 
         [Route("{id}")]
         [HttpPut]
         public async Task<IHttpActionResult> UpdateAsync([FromUri] string id, [FromBody] RestaurantDto restaurant)
         {
-            if (string.IsNullOrEmpty(id) || await DBManager.GetRestaurantWithIdAsync(id) == null || string.IsNullOrEmpty(restaurant?.Name) || (restaurant.MenuLink != null && !Uri.IsWellFormedUriString(restaurant.MenuLink.ToString(), UriKind.Absolute)))
+            if (string.IsNullOrEmpty(id) || await DBManager.GetEntityByIdAsync<Restaurant>(id) == null || string.IsNullOrEmpty(restaurant?.Name) || (restaurant.MenuLink != null && !Uri.IsWellFormedUriString(restaurant.MenuLink.ToString(), UriKind.Absolute)))
             {
                 return BadRequest();
             }
-            return Ok(await DBManager.UpdateRestaurantAsync(id, restaurant));
+            Restaurant r = new Restaurant()
```

Comment:
You are correct that a PUT replaces the whole entity...however, there is a difference between creating a completely new entity and mutating all fields on an existing entity. In this case, we want to mutate all mutable fields for an entity. Let's say we add things like last update date or other audit fields in an entity. We'll never let the API mutate those fields. However a PUT will replace all fields we allow mutations for. So "replace" is only for all mutable fields, and not necessarily a completely different entity. Does that make sense?